src/main/plugin/iso19139.ca.HNAP/schematron/schematron-rules-common.sch

```diff
@@ -57,6 +57,33 @@
     </xsl:for-each>
   </xsl:function>
 
+
+  <xsl:function xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    name="geonet:getDuplicateResources"
+    as="xs:string">
+    <xsl:param name="elements" as="element()*"/>
+
+    <xsl:variable name="duplicateResources">
+      <xsl:for-each-group select="$elements"
+        group-by="concat(gmd:linkage/gmd:URL/text(), '|', tokenize(current-group()[1]/gmd:description/gco:CharacterString/text(), ';')[3])">
```

Comment:
I'm not really sure about this code, you are using a `group-by` to group the elements and using the `current-group()` inside the `group-by` clause at the same time. This is not very orthodox, even if apparently seems to work.